benchmarks/tools/ThumbnailResize.cs

```diff
@@ -0,0 +1,400 @@
+			{
+				thumb.Process(new ImageSharpResizeProcessor(KnownResamplers.Triangle));
+				thumb.Write(output);
+			}
+
+			File.Delete(output);
+		}
+
+		[Benchmark]
+		public void ImageSharp_ThumbnailAndResize_Welch()
+		{
+			using (var image = new RawImage(filePath))
+			using (var thumb = image.UnpackThumbnail())
+			{
+				thumb.Process(new ImageSharpResizeProcessor(KnownResamplers.Welch));
+				thumb.Write(output);
+			}
+
+			File.Delete(output);
+		}
+
+		[Benchmark]
+		public void MagicScaler_ThumbnailAndResize_Wic()
+		{
+			using (var image = new RawImage(filePath))
+			using (var thumb = image.UnpackThumbnail())
+			{
+				thumb.Process(new MagicScalerProcessor(200, 200));
+				thumb.Write(output);
+			}
+
+			File.Delete(output);
+		}
+
+		[Benchmark]
+		public void ImageMagick_ThumbnailAndResize()
+		{
+
+			using (var image = new RawImage(filePath))
+			using (var thumb = image.UnpackThumbnail())
+			{
+				thumb.Process(new ImageMagickResizerProcessor());
+				thumb.Write(output);
+			}
+
+			File.Delete(output);
+		}
+
+		class ImageMagickResizerProcessor : ThumbnailProcessor
+		{
+			byte[] buffer;
+			public override void Process(RawImageData data)
+			{
+				var image = new MagickImage(data.Buffer);
+				image.Resize(200, 200);
+
+				var stream = new MemoryStream();
+				image.Write(stream);
+				buffer = stream.ToArray();
+			}
+
+			public override void Write(RawImageData data, string file)
+			{
+				File.WriteAllBytes(file, buffer);
+			}
+		}
+
+		class ImageSharpResizeProcessor : ThumbnailProcessor
+		{
+			Image image;
+			IResampler resampler;
+			public ImageSharpResizeProcessor(IResampler resampler)
+			{
+				this.resampler = resampler;
+			}
+
+			public override void Process(RawImageData data)
+			{
+				image = Image.Load(data.Buffer);
+				if (image.Width > image.Height)
+					image.Mutate(ctx => ctx.Resize(200, 0, resampler));
+				else
+					image.Mutate(ctx => ctx.Resize(0, 200));
+			}
+
+			public override void Write(RawImageData data, string file)
+			{
+				image.Save(file);
+			}
+
+			protected override void Dispose(bool disposing)
+			{
+				if (IsDisposed)
+					return;
+
+				if (disposing)
+				{
+					if (image != null)
+					{
+						image.Dispose();
+						image = null;
+					}
+				}
+			}
+		}
+
+		class MagicScalerProcessor : ThumbnailProcessor
+		{
+			byte[] buffer;
+			readonly int width, height;
+			public MagicScalerProcessor(int width, int height)
+			{
+				this.width = width;
+				this.height = height;
+			}
+
+			public override void Process(RawImageData data)
+			{
+				using (var stream = new MemoryStream())
+				{
+					var settings = new ProcessImageSettings
+					{
+						Width = this.width
+					};
+
+					MagicImageProcessor.ProcessImage(data.Buffer, stream, settings);
+					buffer = stream.ToArray();
+				}
+			}
+
+			public override void Write(RawImageData data, string file)
+			{
+				File.WriteAllBytes(file, buffer);
+			}
+		}
```

Comment:
@mitchelsellers I created multiple `ThumbnailProcessors` using various image processing libraries to demonstrate the flexibility in this pattern. I don't think we are going to keep this benchmark in the library as we shouldn't be taking other direct dependencies. I wanted here in the PR just to demonstrate how we can easily hook-up different libraries to work with Imaging.Raw